### PR TITLE
Setup initial docker config

### PR DESCRIPTION
**Pull request by toddkblake**
Created both a `Dockerfile` for production use and a `docker-compose` for development use. We can modify / trim down later. But this is a good starting point. Also updated the readme.

One note about the docker build, it uses a two-step build stage approach. The first builds the go binary, but because the image contains all of the `golang` and `dep` source code, it's rather large `~150mb`. So instead, the first stage builds the binary, and then the second stage builds a very minimal `alpine` linux distro and copies the go binary on to it. This gets the image size down to like `~15mb`. Only thing I can't seem to do in the docker file is remove the intermediary image and dependent images (i.e. `golang:alpine`). However, a quick `docker image prune` should do the job.

To Test:
1. Try starting up in dev with `docker-compose up -d`. 
2. Go to `localhost:8080/status` and verify you get an ok response.
3. Shut down the dev server with `docker-compose down`.
4. Try building a production image with `docker build -t s3archiver .`.
5. Start running an s3archiver container with `docker run -i -t -p 8080:8080 s3archiver`.
6. Go to `localhost:8080/status` and verify you get an ok response.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+# Download and install the latest release of dep
+ADD https://github.com/golang/dep/releases/download/v0.4.1/dep-linux-amd64 /usr/bin/dep
+RUN chmod +x /usr/bin/dep
+
+# Install git in order to install dependencies
+RUN apk add --no-cache git mercurial
+
+# Install dependencies and build the binary from the src
+WORKDIR $GOPATH/src/github.com/funk-palace/s3archiver
```
Comment (KSauri) [inline Dockerfile]: Is this a convention for golang?  I'm assuming it is because you usually run `go get <github url>`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+COPY Gopkg.toml Gopkg.lock ./
+RUN dep ensure --vendor-only
+COPY . ./
+RUN go build -o /s3archiver server.go
+
+## Second build stage to copy binary to final image
+FROM alpine
+COPY --from=build /s3archiver ./
+ENTRYPOINT ./s3archiver
```
Comment (KSauri) [inline Dockerfile]: do we want to use `CMD` here to optionally give the container S3 variables?

File: Dockerfile (hunk 1 above)
Comment (toddkblake) [reply Dockerfile]: @KSauri yeah, basically when you run `go get` it always put's the package in this directory structure.

File: Dockerfile (hunk 2 above)
Comment (toddkblake) [reply Dockerfile]: @KSauri I want to add on the s3 config stuff later cause i'm not exactly sure what we need yet and how to do it. But I'm thinking you should be able to do it either way

**Pull request merged**